### PR TITLE
Change the name of method getQueryTableName and remove its $query argument

### LINKED ISSUE #12878: Expose ActiveQuery->getQueryTableName()
Can this method be exposed so it can be used in extended `ActiveQuery` classes.
Would allow you to easily use table aliases.

```php
class ExtraQuery extends ActiveQuery
{
    //...
    public function extraColumn()
    {
        list(, $tableName) = $this->getQueryTableName($this);
        $this->addSelect("$tableName.column");
        return $this;
    }
    //...
}
```

### PR DESCRIPTION
**Pull request by dynasource**
see the descriptions of the different commits.
I also fixed consistency for some other unit tests

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: tests/framework/db/ActiveQueryTest.php [hunk 1]
```diff
@@ -129,6 +129,24 @@ public function testInnerJoinWith()
         ], $result->joinWith);
     }
 
+    /**
+     * @todo: tests for the regex inside getQueryTableName
+     */
+    public function testGetQueryTableName_from_not_set()
```
Comment (klimov-paul) [inline tests/framework/db/ActiveQueryTest.php]: Why methods are named not in 'camelCase' ?

**Review by klimov-paul — CHANGES_REQUESTED** [review]

File: tests/framework/db/ActiveQueryTest.php (hunk 1 above)
Comment (samdark) [inline tests/framework/db/ActiveQueryTest.php]: Yes, should be renamed.

**Comment by klimov-paul** [comment]
Actually the entire change does not feels right. In the scope of `ActiveQuery` method `getQueryTableName()` seems to be ambiguous: the table name, which query is acciciated to is already covered by `Query::from`.
Originally method `getQueryTableName()` was an internal helper for the query bulding process and thus its presence was OK. But this PR changes it purpose creating a twisted public interface.

If you wish `ActiveQuery` to be able to return a related table name you should convert `from` into a virtual property with getter, which can prefill it by `ActiveRecord::tableName()` if its value is empty.

File: tests/framework/db/ActiveQueryTest.php (hunk 1 above)
Comment (dynasource) [inline tests/framework/db/ActiveQueryTest.php]: for the simple reason that multiple tests for the same method can be easily distinguished

Comment (dynasource) [inline tests/framework/db/ActiveQueryTest.php]: an alternative could be testGetQueryTableName_fromNotSet(), its a matter of taste and the depth in which you are testing. If you have 20 different test methods with similarities, the underscores help significantly with an easy comparison

Comment (klimov-paul) [inline tests/framework/db/ActiveQueryTest.php]: Does not sound convincing to me

**Comment by dynasource** [comment]
I totally disagree. The current implementation shows zero reason for having $query as an argument.

**Comment by klimov-paul** [comment]
???
So do you think the argument should be removed entirely then?
What about these places:

https://github.com/yiisoft/yii2/blob/master/framework/db/ActiveQuery.php#L606
https://github.com/yiisoft/yii2/blob/master/framework/db/ActiveQuery.php#L607

**Comment by samdark** [comment]
Hmm... what @klimov-paul says makes sense. At least problem part of it.

**Comment by dynasource** [comment]
I dont see a $this inside getQueryTableName(), then it makes no sense to have $query as an argument. Its confusing.

**Comment by samdark** [comment]
@dynasource see lines @klimov-paul linked to.

**Review by samdark — CHANGES_REQUESTED** [review]
Seems it won't work this way.

**Review by samdark — APPROVED** [review]
No, that's OK :)

**Comment by samdark** [comment]
Re-checked. Changes looks OK to me.

File: tests/framework/db/ActiveQueryTest.php (hunk 1 above)
Comment (dynasource) [inline tests/framework/db/ActiveQueryTest.php]: have you every created 20 test methods to test 1 PHP function?

**Comment by samdark** [comment]
@klimov-paul the places you've referred to are covered.

**Comment by klimov-paul** [comment]
I will not accept this change unless @cebe agreed on it.

**Comment by samdark** [comment]
#12878

File: tests/framework/db/ActiveQueryTest.php (hunk 1 above)
Comment (dynasource) [inline tests/framework/db/ActiveQueryTest.php]: for normal framework code, yes. camelCase will do. 
But not if you are testing:
* 1 method:
* on multiple criteria
* in multiple test methods (to keep the smallest contained environment + to have an instant understanding if a test fails)

Comment (dynasource) [inline tests/framework/db/ActiveQueryTest.php]: https://books.google.nl/books?id=K5dYi2HfAOYC&pg=PA82&lpg=PA82&dq=phpunit+methods+underscore+distinguish&source=bl&ots=L_aiHseLUp&sig=qk_bWtDkN7yk17oDjgeZC_hZ6KU&hl=nl&sa=X&ved=0ahUKEwiModPdjorQAhXKCywKHReLAwoQ6AEIUDAG#v=onepage&q=phpunit%20methods%20underscore%20distinguish&f=false

**Review by cebe — CHANGES_REQUESTED** [review]
we may turn it into a protected method as it was discussed in #12878 but it should not be public. Public method will add something to the interface that does not belong here. All other methods are for constucting/modifying the Query object. Only this one would be a getter, this creates inconsistency in the interface.

**Comment by dynasource** [comment]
the function name should also be altered. The current PR still shows the old name 'getQueryTableName' and without $query as an argument, it should at least be named getTableName(). But, regarding all of our conversations, this name is also not preferable. @klimov-paul stated that this method is meant for internal processing. It should get a name that clarifies that.

**Comment by samdark** [comment]
Agree.

**Comment by cebe** [comment]
I have reviewed the method now in more detail and I think it should not be added to the public interface, not even protected.

- Its original implementation was to determine table names from `from` part in certain circumstances. This works as expected for the code where it is used now and appropriate checks are made, e.g. in `alias()` ,method. It may not work as expected, when `from` contains multiple tables.
- If we want to provide a new method for retrieving an alias or table name, we get to the same discussion we already have/had in #9326 and #7263.
- there should be a `getAlias()` and/or `getTableName()` method whos implementation may be based on this internal method, but that should be discussed in the issues and addressed separately from the refactoring made here.

additionally to `from`, which is already there

**Comment by dynasource** [comment]
I agree. Do you have a suggestion for the private method to be newly called?

**Comment by samdark** [comment]
`getTableNameAndAlias()`?

**Comment by dynasource** [comment]
'getTableNameAndAlias' is the most logical with respect to the objective of the method. But we should also take the 'temporary processing' aspect into account. Couldnt we add add the '@internal' tag to this method? Based on the PHPdoc specs https://phpdoc.org/docs/latest/references/phpdoc/tags/internal.html, this tag:
* The @internal tag can be used as counterpart of the @api tag, indicating that the associated Structural Elements are used purely for the internal workings of this piece of software.

This would saves us from confusion in the future

**Comment by cebe** [comment]
a private method is "internal" anyway so there is no need for such a label.

**Comment by dynasource** [comment]
of course, but it helps emphasizing that it should only be used internally and never be made public. It would have prevented this whole discussion and the issue at https://github.com/yiisoft/yii2/issues/12878

**Comment by samdark** [comment]
Name should be changed in the rest of the code, it seems...

**Comment by dynasource** [comment]
good to see my added tests in action ;)

**Review by cebe — APPROVED** [review]

**Comment by cebe** [comment]
Thank you!

**Pull request merged**